src/pkg/selectors/exchange_test.go

```diff
@@ -1041,6 +1041,233 @@ func (suite *ExchangeSelectorSuite) TestExchangeRestore_Reduce() {
 	}
 }
 
+func (suite *ExchangeSelectorSuite) TestExchangeRestore_Reduce_locationRef() {
```

Comment:
I understand that this test showcases how things will work when we can get everything switched over to storing the LocationRef. However, we're not currently in that situation so I think it would be best to have this test ensure that email and contacts selectors continue to properly compare the path elements in the RepoRef while the events selector compares the path elements in the LocationRef

Since I think we currently lack the sort of e2e tests for the above I think this is the best we can do to make sure things work as expected in the broader system

Reply:
There's a catch here: this code doesn't really recognize expectations like that from the broader system.  I could comment out the mail and event tests here, which is probably sufficient for the time being.  But the reality of the selectors code is that it doesn't (and shouldn't) evaluate whether or not the thing is logically supported as a use case.  It only cares that it can run the provided data correctly.  IE: this test isn't meant to show that we _are_ handling locations for mail and contacts, just that we _can_.

I could add a comment explaining this, if that helps?

Reply:
mmmm logically I think that explanation makes sense. I think it still leaves a hole in our testing where we can't actually say that these changes don't break the current functionality for email and contacts because we have no tests ensuring it works as expected

Reply:
I think we'll have to go elsewhere for that.  Probably at the collections, to enforce that the location isn't populated for email and contacts.

Also, it should just work, even if things slip through?  Because that value _should_ be a copy of our the current folders for those types.

Reply:
mmm I suppose that's true. Just not having automated tests for system-wide invariants makes me a bit nervous because it's easy to forget to check them 😅